src/Board.py

```diff
@@ -52,6 +58,18 @@ def __init__(self, style: str = "default", size: int = 5) -> None:
         else:
             raise ValueError("Unrecognised map style")
 
+    def start(self, terrain_filter: Terrain | None = None):
```

Comment:
It may be used for finding start positions but might also be useful in other situations, so having a more generic name might be helpful. Perhaps `get_tiles_of_terrain`? Also I can't think of a time when this would need to be called with a `None` value for terrain filter so maybe we don't need to add a case for that?